### DISCUSSION TITLE
@InheritConfiguration method lookup through the @Mapper(uses) field?

### DISCUSSION THREAD

**Discussion by RichMacDonald** [mapstruct/mapstruct#3194] [General]
Greetings. I asked this question over at the google groups thread and have been wondering why no one in the last year seems to have any interest in the project, lol. Luckily, I finally found this destination.

The documentation states that @InheritConfiguration method lookup is "in the current mapper, a super class/interface, or in the shared configuration interface".

Is it feasible (what will it take) to extend method lookup to include the @Mapper(uses = "OtherClasses") classes?

My specific interest is mapping a class hierarchy with 1K+ classes. Unwieldly to put everything into a single file. I hoped I could define the mapping for a specific subclass in one file then "refer" to the mapping for the superclass via the @Mapper(uses = "TheSuperClassMapper"), but this is not supported. My current solution is to use the "experimental" Mapping Composition. Each class being mapped has an equivalent MappingCompositionAnnotation and I "inherit" that way. It works fine.

**Comment by filiphr** [comment]
>Greetings. I asked this question over at the google groups thread and have been wondering why no one in the last year seems to have any interest in the project, lol. Luckily, I finally found this destination.

Sorry for that. I've updated the Google Groups to have a link to the GitHub discussions.

>Is it feasible (what will it take) to extend method lookup to include the `@Mapper(uses = "OtherClasses")` classes?

This would be a rather complex thing to do, since it might lead to a lot of backwards incompatible problems.

What we could accept is to have a way to define multiple configurations that you might be able to use. However, we have to thing about merging those.

>My current solution is to use the "experimental" Mapping Composition. Each class being mapped has an equivalent MappingCompositionAnnotation and I "inherit" that way. It works fine.

I would say that this experimental Mapping Composition is way better alternative than the `@InheritConfiguration`. So I would go with that, even though we say it is experimental. We've named it like that in order to have some leeway in case we needed to change something. However, I think that we can remove that now. I've created #3239 for this
